library/src/main/java/com/cube/storm/ui/lib/adapter/StormSpotlightAdapter.java

```diff
@@ -95,10 +98,31 @@ public Object instantiateItem(@NonNull ViewGroup container, int position)
 		final SpotlightImageProperty spotlightItem = getItem(position);
 		if (spotlightItem != null)
 		{
+			View textContainer = view.findViewById(R.id.text_container);
 			ImageView imageView = view.findViewById(R.id.image_view);
-			TextView textView = view.findViewById(R.id.text_ticker);
+			TextView category = view.findViewById(R.id.category);
+			TextView title = view.findViewById(R.id.title);
+			TextView description = view.findViewById(R.id.description);
+
 			imageView.populate(spotlightItem.getImage());
-			textView.populate(spotlightItem.getText(), spotlightItem.getLink());
+			category.populate(spotlightItem.getCategory());
+			title.populate(spotlightItem.getTitle());
+			description.populate(spotlightItem.getDescription());
+
+			// Hide text container under spotlight image if no text to populate
+			Processor<TextProperty, String> textProcessor = UiSettings.getInstance().getTextProcessor();
```

Comment:
No need to use the text processor here. The TextViews above are already showing / hiding themselves based on whether there is text. So instead you can just use category.visibility == VISIBLE etc.